Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs

```diff
@@ -511,5 +519,283 @@ public static CairoSettingsWindow Instance
+                " *.mpg; *.mpv2; *.mts; *.nsv; *.nuv; *.ogg; *.ogm; *.ogv; *.ogx; *.ps;" +
+                " *.rec; *.rm; *.rmvb; *.tod; *.ts; *.tts; *.vob; *.vro; *.webm | DreamScene | *.zip";
+        }
+
+        private void cboDesktopBackgroundType_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (cboDesktopBackgroundType.SelectedItem != null)
+            {
+                ComboBoxItem item = cboDesktopBackgroundType.SelectedItem as ComboBoxItem;
+                windowsImageBackgroundStackPanel.Visibility = (item.Tag == windowsImageBackgroundStackPanel) ? Visibility.Visible : Visibility.Collapsed;
+                cairoImageBackgroundStackPanel.Visibility = (item.Tag == cairoImageBackgroundStackPanel) ? Visibility.Visible : Visibility.Collapsed;
+                cairoVideoBackgroundStackPanel.Visibility = (item.Tag == cairoVideoBackgroundStackPanel) ? Visibility.Visible : Visibility.Collapsed;
+                bingImageBackgroundStackPanel.Visibility = (item.Tag == bingImageBackgroundStackPanel) ? Visibility.Visible : Visibility.Collapsed;
+
+                Settings.Instance.DesktopBackgroundType = item.Name;
+            }
+        }
+
+        #endregion
+
+        private void cboWindowsBackgroundStyle_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            // Stretched { WallpaperStyle = 2; TileWallpaper = 0 }
+            string wallpaperStyle = "2";
+            string tileWallpaper = "0";
+            switch (cboBingBackgroundStyle.SelectedItem)
+            {
+                case Desktop.CairoWallpaperStyle.Tile: // Tiled { WallpaperStyle = 0; TileWallpaper = 1 }
+                    wallpaperStyle = "0";
+                    tileWallpaper = "1";
+                    break;
+                case Desktop.CairoWallpaperStyle.Center: // Centered { WallpaperStyle = 1; TileWallpaper = 0 }
+                    wallpaperStyle = "1";
+                    tileWallpaper = "0";
+                    break;
+                case Desktop.CairoWallpaperStyle.Fit: // Fit { WallpaperStyle = 6; TileWallpaper = 0 }
+                    wallpaperStyle = "6";
+                    tileWallpaper = "0";
+                    break;
+                case Desktop.CairoWallpaperStyle.Fill: // Fill { WallpaperStyle = 10; TileWallpaper = 0 }
+                    wallpaperStyle = "10";
+                    tileWallpaper = "0";
+                    break;
+                case Desktop.CairoWallpaperStyle.Span: // Span { WallpaperStyle = 10; TileWallpaper = 0 }
+                    wallpaperStyle = "20";
+                    tileWallpaper = "0";
+                    break;
+            }
+
+            Registry.GetValue(@"HKEY_CURRENT_USER\Control Panel\Desktop", "WallpaperStyle", wallpaperStyle);
+            Registry.GetValue(@"HKEY_CURRENT_USER\Control Panel\Desktop", "TileWallpaper", tileWallpaper);
```

Comment:
Shouldn't these two be Registry.SetValue ?

Reply:
Ohhhh, good catch. The ComboBox used in the switch was incorrect too.